csvy/readers.py

```diff
@@ -270,3 +270,30 @@ def read_to_list(
             data.append(row)
 
     return data, header
+
+
+def basic_read(
+    filename: Path | str, marker: str = "---", **kwargs: Any
+) -> tuple[dict[str, Any], dict[str, Any]]:
+    """Reads a CSVY file into a dictionary with 'columns' and 'data'.
+
+    Args:
+        filename: Name of the file to read.
+        marker: The marker characters that indicate the YAML header.
+        **kwargs: Additional arguments for YAML parsing.
+
+    Returns:
+        Tuple containing:
+            - A dictionary with 'columns' (list of column names) and
+              'data' (2D list of row data).
+            - A dictionary with metadata from the header.
+    """
+    data, metadata = read_to_list(filename, marker=marker, **kwargs)
+
+    # Clean column names and data rows
+    columns = [col.strip() for col in data[0]] if data else []
+    row_data = (
+        [[cell.strip() for cell in row] for row in data[1:]] if len(data) > 1 else []
+    )
+
+    return {"columns": columns, "data": row_data}, metadata
```

Comment:
It feels rather odd to have the column names but then the data provided as rows. There's a complete disconnection between both.

I think it will make more sense to provide the data as columns, rather than rows - and I think that even better if in a dictionary form, such that the keys are the column names. Something like:

```python
datadict= {
    data[0][0]: data[0][1:],
    data[1][0]: data[1][1:],
    ...
}
```
Although I'm not sure if this is what @AdrianDAlessandro had in mind in https://github.com/ImperialCollegeLondon/pycsvy/issues/91

For this to work, the data needs to be column wise rather than row wise. That can be done using the `zip_longest` from the `itertools` library:

```python
data = list(map(list, zip_longest(*data, fillvalue=fillvalue)))
```
With `fillvalue` a value to populate the rows with if they have different length. Now you can create the `datadict` above using the first element of each column as the key.

With these approach, I think this function will be rather powerful.

Reply:
Yes, exactly. Column headers should be the dictionary keys, and the data in each column should be a list